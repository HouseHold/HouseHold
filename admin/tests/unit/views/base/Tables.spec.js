import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue'
import Tables from '@/views/base/Tables'

Vue.use(CoreuiVue)

describe('Tables.vue', () => {
  it('has a name', () => {
    expect(Tables.name).toBe('Tables')
  })
  it('is Vue instance', () => {
    const wrapper = mount(Tables)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Tables', () => {
    const wrapper = mount(Tables)
    expect(wrapper.is(Tables)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Tables)
    expect(wrapper.element).toMatchSnapshot()
  })
})
