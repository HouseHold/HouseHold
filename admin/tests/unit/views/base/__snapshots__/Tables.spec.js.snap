// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tables.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="6"
      tag="div"
    >
      <ctablewrapper-stub
        caption="Table"
        fields="username,registered,role,status"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      tag="div"
    >
      <ctablewrapper-stub
        caption="Striped Table"
        fields="username,registered,role,status"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        striped="true"
      />
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="6"
      tag="div"
    >
      <ctablewrapper-stub
        caption="Condensed Table"
        fields="username,registered,role,status"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        small="true"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      tag="div"
    >
      <ctablewrapper-stub
        bordered="true"
        caption="Bordered Table"
        fields="username,registered,role,status"
        fixed="true"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      />
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      sm="12"
      tag="div"
    >
      <ctablewrapper-stub
        bordered="true"
        caption="Combined All Table"
        fields="username,registered,role,status"
        fixed="true"
        hover="true"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        small="true"
        striped="true"
      />
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      sm="12"
      tag="div"
    >
      <ctablewrapper-stub
        bordered="true"
        caption="Combined All dark Table"
        dark="true"
        fields="username,registered,role,status"
        fixed="true"
        hover="true"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        small="true"
        striped="true"
      />
    </ccol-stub>
  </crow-stub>
</div>
`;
